### DISCUSSION TITLE
Is there an equivalent to the External Tools menu system found in Visual Studio?

### DISCUSSION THREAD

**Discussion by g40** [eranif/codelite#3522] [Q&A] [answered]
I ask as it is incredibly useful for building `makefile' based projects that are other painful to import. One would generally add an entry to build and another to clean.
And used together with the wonderful quick debug facility it means Codelite can be productive in seconds.
Thanks for listening. Codelite is a fantastic resource.

**Comment by eranif** [comment, marked as answer]
There are 2 options:

- Use the "External tools" plugin
- Use a "File System Workspace" **recommended**

The File-System-Workspace is basically: point CodeLite to a folder and it will use it as the root of the workspace.
In addition, the File-System-Workspace has a settings dialog where you can set custom build commands. In fact, this is how I build CodeLite.

See here:

https://docs.codelite.org/workspaces/file_system/

**Reply by g40** [reply]
Hello @eranif 

Thank you. I just discovered the plugins - they do not get installed by default if you use `apt` on Debian/Ubuntu.

Things seem to be working - `Tool Path` though might be better described as `Command line`
